### ISSUE TITLE
Unable to debug `NcPlane::gradient`

### ISSUE THREAD

**Issue by CyanideForBreakfast** [dankamongmen/libnotcurses-sys#19]
I've been meaning to draw empty cells across a line, yet am unable to debug why `gradient` function is throwing error.
Here's a simple reproducible example.
```
use libnotcurses_sys::*;

fn main() -> NcResult<()> {
    let nc = unsafe { Nc::with_flags(NcOptions::SUPPRESS_BANNERS)? };
    let splane = unsafe { nc.stdplane() };

    let header_bg_channel = NcChannel::from_rgb8(
            57,
            100,
            200,
        );
    let header_fg_channel = NcChannel::from_rgb8(
            200,
            200, 
            200
        );

    let header_combined_channel = NcChannels::combine(header_fg_channel, header_bg_channel);
    splane.gradient(
        Some(0),
        Some(0),
        Some(1),
        None,
        " ",
        0,
        header_combined_channel,
        header_combined_channel,
        header_combined_channel,
        header_combined_channel,
    )?;
    nc.render()?;
    Ok(())
}
```
In some rare cases, it does render with garbage values filled in. Am I doing something wrong or making some invalid assumptions?

**OS**: Arch
**Terminals**: Xterm, Alacritty, Kitty

**Comment by joseluis** [comment]
Thanks for the bug report, I suspect this may be related to recent updates to the static channels functions in the C library. I'll try to update the Rust bindings and fix any related bugs.

**Comment by joseluis** [comment]
It turns out it wasn't that.

The good news is that I've reproduced your issue, so it's an implementation bug. I'll investigate it further and will attempt to fix it soon.

BTW are you running an older version of the library? I've updated your code to the last version pointing the differences:
```rust
use libnotcurses_sys::*;

fn main() -> NcResult<()> {
    let nc = unsafe { Nc::with_flags(NcFlags::SuppressBanners))? }; // <=

    let splane = unsafe { nc.stdplane() };

    let header_bg_channel = NcChannel::from_rgb8(
            57,
            100,
            200,
        );
    let header_fg_channel = NcChannel::from_rgb8(
            200,
            200, 
            200
        );

    let header_combined_channel = NcChannels::combine(header_fg_channel, header_bg_channel);
    splane.gradient(
        Some(0),
        Some(0),
        Some(1),
        None,
        " ",
        NcStyle::None, // <=
        header_combined_channel,
        header_combined_channel,
        header_combined_channel,
        header_combined_channel,
    )?;
    nc.render()?;
    unsafe { nc.stop()?; } // <= notcurses must be stopped before exiting or it will leak
    Ok(())
}
```

**Comment by CyanideForBreakfast** [comment]
Yes, I'm currently fixed on `libnotcurses - 3.1.2`. The `3.2.0` version just wouldn't compile when I put it under `[dependencies]` in my `Cargo.toml`, spitting out a bunch of errors. On my local Arch system, I've already installed the [notcurses-git AUR](https://aur.archlinux.org/packages/notcurses-git). I thought that there must be a version discrepancy between the AUR and the version `3.2.0` depends on that is causing the errors, so decided maybe I'll switch in the future. Is that related?

**Comment by joseluis** [comment]
I've updated version 3.3.0 that should have the gradient problems fixed.

You'll have to update to the new syntax, though.
